src/main/java/org/sharedid/endpoint/handler/CheckOptOutHandler.java

```diff
@@ -0,0 +1,61 @@
+package org.sharedid.endpoint.handler;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.Cookie;
+import io.vertx.ext.web.RoutingContext;
+import org.sharedid.endpoint.context.DataContext;
+import org.sharedid.endpoint.util.ExtraHeadersCookie;
+import org.sharedid.endpoint.util.ResponseUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CheckOptOutHandler implements Handler<RoutingContext> {
+    private static final Logger logger = LoggerFactory.getLogger(CheckOptOutHandler.class);
+
+    private String optOutCookieValue;
+    private String sharedIdCookieName;
+    private Long sharedIdOptOutTtl;
+    private Boolean isSecureCookiesEnabled;
+
+    @Autowired
+    public CheckOptOutHandler(@Value("${cookie.shared-id.opt-out-value}") String optOutCookieValue,
+                              @Value("${cookie.shared-id.name}") String sharedIdCookieName,
+                              @Value("${cookie.shared-id.opt-out-ttl}") Long sharedIdOptOutTtl,
+                              @Value("${cookies.secure}") Boolean isSecureCookiesEnabled) {
+        this.optOutCookieValue = optOutCookieValue;
+        this.sharedIdCookieName = sharedIdCookieName;
+        this.sharedIdOptOutTtl = sharedIdOptOutTtl;
+        this.isSecureCookiesEnabled = isSecureCookiesEnabled;
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        DataContext dataContext = DataContext.from(routingContext);
+
+        String userId = dataContext.getUserId();
+
+        boolean hasOptedOutUserId = optOutCookieValue.equals(userId);
+
+        if (hasOptedOutUserId || dataContext.getHasOptOutCookie()) {
+            logger.debug("Is opted out");
+            //update opt out cookie ttl
+            updateOptOutCookie(routingContext);
```

Comment:
Since the code calls `updateOptOutCookie` for opt out user, and inside `updateOptOutCookie` it calls `responseCookie.setMaxAge`, which means for opt out user, every time it calls `CheckOptOutHandler`, cookie `"${cookie.shared-id.name}"` with `${cookie.shared-id.opt-out-value}` value will extent its expiry. It doesn't look correct to me.

Reply:
We need to extend the age of the opt out cookie otherwise it will eventually expire, which to us looks like an opt in which was not the users's intention (opt out is forever, normal cookies expire).

Reply:
Got it. So keep opt out option forever is by design.
Thanks for information.